src/wallet/fees.cpp

```diff
@@ -15,10 +16,9 @@ CAmount GetRequiredFee(const CWallet& wallet, unsigned int nTxBytes)
     return GetRequiredFeeRate(wallet).GetFee(nTxBytes);
 }
 
-
 CAmount GetMinimumFee(const CWallet& wallet, unsigned int nTxBytes, const CCoinControl& coin_control, FeeCalculation* feeCalc)
 {
-    return GetMinimumFeeRate(wallet, coin_control, feeCalc).GetFee(nTxBytes);
+    return GetMinimumFeeRate(wallet, coin_control, feeCalc).GetFee(nTxBytes, coin_control.m_fee_asset.value_or(::policyAsset));
```

Comment:
`GetMinimumFeeRate` is where `mintxfee`, `paytxfee`, and
 `blockmintxfee` are used to negotiate a suitable fee rate for the transaction, and passing the asset parameter here ensures that the result is denominated in the fee asset and not RFU.